src/wp-includes/class-wp-autoload.php

```diff
@@ -0,0 +1,517 @@
+		'wp_terms_list_table'                         => 'wp-admin/includes/class-wp-terms-list-table.php',
+		'wp_theme_install_list_table'                 => 'wp-admin/includes/class-wp-theme-install-list-table.php',
+		'wp_themes_list_table'                        => 'wp-admin/includes/class-wp-themes-list-table.php',
+		'wp_upgrader_skin'                            => 'wp-admin/includes/class-wp-upgrader-skin.php',
+		'wp_upgrader'                                 => 'wp-admin/includes/class-wp-upgrader.php',
+		'wp_users_list_table'                         => 'wp-admin/includes/class-wp-users-list-table.php',
+
+		/* Classes in wp-admin/includes/deprecated.php. */
+		'wp_user_search'                              => 'wp-admin/includes/deprecated.php',
+		'wp_privacy_data_export_requests_table'       => 'wp-admin/includes/deprecated.php',
+		'wp_privacy_data_removal_requests_table'      => 'wp-admin/includes/deprecated.php',
+	);
+
+	/**
+	 * Whether the autoloader has already been registered or not.
+	 *
+	 * Avoid registering the autoloader multiple times.
+	 *
+	 * @static
+	 * @access private
+	 *
+	 * @var bool
+	 */
+	private static $registered = false;
+
+	/**
+	 * Register the autoloader.
+	 *
+	 * @return void
+	 */
+	public static function register() {
+		// Bail early if already registered.
+		if ( self::$registered ) {
+			return;
+		}
+
+		self::register_external_bundled();
+		self::register_core();
+
+		self::$registered = true;
+	}
+
+	/**
+	 * Register the autoloader for external, bundled libraries.
+	 *
+	 * @return void
+	 */
+	public static function register_external_bundled() {
+		require_once ABSPATH . 'wp-includes/Requests/src/Autoload.php';
+		require_once ABSPATH . 'wp-includes/sodium_compat/autoload.php';
+
+		spl_autoload_register( array( '\WpOrg\Requests\Autoload', 'load' ) );
+	}
+
+	/**
+	 * Register the autoloader for WordPress Core classes.
+	 *
+	 * @return void
+	 */
+	public static function register_core() {
+		spl_autoload_register( array( __CLASS__, 'autoload_core' ), true, true );
+	}
+
+	/**
+	 * Autoload a WordPress class.
+	 *
+	 * @param string $class_name Class name.
+	 * @return bool True if the class was loaded, false otherwise.
+	 */
+	public static function autoload_core( $class_name ) {
+		// Lowercase the classname to accommodate for WP classes written with wrong cases.
+		$class_name = strtolower( $class_name );
+
+		// Bail early if the class is not a WP class.
+		// Use empty() instead of !isset() for performance reasons (saves a BOOL_NOT opcode).
+		if ( empty( self::CLASSES_PATHS[ $class_name ] ) ) {
+			return false;
+		}
+
+		/*
+		 * SimplePie classes should additionally load the `wp-includes/class-simplepie.php` file
+		 * to ensure that constants are defined.
+		 */
+		if ( str_starts_with( $class_name, 'simplepie' ) ) {
+			require_once ABSPATH . self::CLASSES_PATHS['simplepie'];
+		}
+
+		require_once ABSPATH . self::CLASSES_PATHS[ $class_name ];
+		return true;
```

Comment:
I'm assuming this would also mean we should remove the SimplePie classes from the classMap, right? https://github.com/WordPress/wordpress-develop/blob/55d2f04e720ba192589edf3e8119535e64858264/src/wp-includes/class-wp-autoload.php#L231-L264

Reply:
Right now, yes, as it looks like the `wp-includes/class-simplepie.php` file then automatically requires all other classes: https://github.com/WordPress/wordpress-develop/blob/2ae28c50f9e97fcf96205ddd8597ceb8892f70ed/src/wp-includes/class-simplepie.php#L5-L17.

Alternatively, we could rewrite SimplePie to use proper autoloading as well, but I think that should be out of the scope of the current PR at this point.

Reply:
@schlessera I pushed some tweaks for SimplePie:
I Removed the SimplePie classes from the classMap, and changed `require_once` to `require`, adding the early bail you posted above.
But then, reading the contents of `class-simplepie.php` I saw that it uses `require` to load all the other files. So that file should be included only once, otherwise we'll get PHP errors. 
One option would be to add a static var in the object to determine if SImplePie has been loaded so we can conditionally load it... But that would defeat the purpose of using `require` instead of `require_once`. With that in mind, I used `require_once` again to load SimplePie 🤷 
I changed the other `require_once` to `require` for files that are in the classMap, but for SimplePie specifically I think we should leave it as `require_once` 🤔 

SimplePie is only loaded in `fetch_feed`, so I think we'll be OK using `require_once` instead of `require` for that one...